crates/notary/server/src/server.rs

```diff
@@ -59,22 +59,26 @@ pub async fn run_server(config: &NotaryServerProperties) -> Result<(), NotarySer
         debug!("Skipping TLS setup as it is turned off.");
         None
     } else {
-        let (tls_private_key, tls_certificates) = load_tls_key_and_cert(
-            &config.tls.private_key_pem_path,
-            &config.tls.certificate_pem_path,
-        )
-        .await?;
-
-        let mut server_config = ServerConfig::builder()
-            .with_safe_defaults()
-            .with_no_client_auth()
-            .with_single_cert(tls_certificates, tls_private_key)
-            .map_err(|err| eyre!("Failed to instantiate notary server tls config: {err}"))?;
-
-        // Set the http protocols we support
-        server_config.alpn_protocols = vec![b"http/1.1".to_vec()];
-        let tls_config = Arc::new(server_config);
-        Some(TlsAcceptor::from(tls_config))
+        if let (Some(private_key_path), Some(certificate_pem_path)) = (
```

Comment:
if the pem paths are not set when `tls.enabled` is true, we should throw an error — like your line 297

Reply:
we have updated pr to be based on the dev branch instead of the main branch and added validation to throw an error when TLS is enabled but PEM paths are not set.

Reply:
@yorozunouchu i didn't see the error throwing fix yet, have you guys pushed it? right now if paths are not set it's still returning `None`, but we need it to throw error — so instead of doing `if let Some(path)...` just directly do something like `path.ok_or_else(...)?`